### PR TITLE
lesson 10 for loading a game world

### PR DESCRIPTION
**Pull request by jianhuang01**
Change webgl [lesson 10](http://learningwebgl.com/lessons/lesson10/index.html) example to use luma.gl API, primary functionality is that loading a texture and a world model to mimic doom game, user can leverage WSAD or cursor keys to move around, page up/down keys to look up and down.

### CODE REVIEW

File: examples/lessons/10/app.js [hunk 1]
```diff
@@ -0,0 +1,157 @@
...
+    });
+    handleKeys();
+    animate();
+  }
+});
+
+animationLoop.getInfo = () => {
+  return `
+  <p>
+    <a href="http://learningwebgl.com/blog/?p=1008" target="_blank">
```
Comment (ibgreen) [inline examples/lessons/10/app.js]: This text and link should be updated?

File: examples/lessons/10/app.js [hunk 2]
```diff
@@ -0,0 +1,157 @@
...
+		  });
+    });
+  },
+  onRender: ({
+    gl, tick, aspect, world
+  }) => {
+    // Update Camera Position
+    const eyePos = [xPos, yPos, zPos];
+    const centerPos = new Matrix4()
+      .identity()
```
Comment (ibgreen) [inline examples/lessons/10/app.js]: We should be able to drop `.identity()`

File: examples/lessons/10/app.js [hunk 3]
```diff
@@ -0,0 +1,157 @@
...
+      clearColor: [0, 0, 0, 1],
+      clearDepth: 1,
+      depthTest: true
+    });
+
+    return loadTextures(gl, {
+      urls: ['mud.gif']
+    })
+    .then(textures => {
+		  return loadFile({url: 'world.txt'})
```
Comment (ibgreen) [inline examples/lessons/10/app.js]: The indentation looks bad here. We use 2 spaces for indent, no tabs. 

Didn't the linter complain in the pre-commit hook?

File: examples/lessons/10/app.js [hunk 4]
```diff
@@ -0,0 +1,157 @@
...
+  resetParameters, setParameters
+} from 'luma.gl';
+
+import {
+  Matrix4
+} from 'math.gl'
+
+import {loadWorldGeometry, World} from './world';
+
+var pitch = 0;
```
Comment (ibgreen) [inline examples/lessons/10/app.js]: We use `let` instead of `var`. Run `npm run lint`

Comment (ibgreen) [reply examples/lessons/10/app.js]: Maybe nice to collect these in an object (e.g. `Player`, or `View` etc)

File: examples/lessons/10/package.json [hunk 5]
```diff
@@ -0,0 +1,19 @@
...
+  "description": "An es6 shadowmap example for luma.gl.",
+  "contributors": [
+    "Rye Terrell <ryeterrell@ryeterrell.net"
+  ],
+  "scripts": {
+    "start": "webpack-dev-server --progress --hot --open -d",
+    "start-local": "webpack-dev-server --env.local --progress --hot --open -d"
+  },
+  "dependencies": {
+    "luma.gl": ">=4.0.0-beta.6",
```
Comment (ibgreen) [inline examples/lessons/10/package.json]: luma.gl 4 is published now, we should use the official version.

File: examples/lessons/10/app.js (hunk 4 above)
Comment (jianhuang01) [reply examples/lessons/10/app.js]: Done both, my Safari(version 10.0.0) doesn't support let and const well, which made this example can't run on Safari, but works well on Chrome, will create an ticket as we discussed offline.

File: examples/lessons/10/app.js (hunk 3 above)
Comment (jianhuang01) [reply examples/lessons/10/app.js]: Done. My editor has tab/space issue, formatted all the text, thanks for reminding. I can't make the linter work on my computer for now.

File: examples/lessons/10/app.js (hunk 1 above)
Comment (jianhuang01) [reply examples/lessons/10/app.js]: This is intended I think cause in luma.gl's example web page we need show the link to webgl lesson pages.

File: examples/lessons/10/package.json (hunk 5 above)
Comment (jianhuang01) [reply examples/lessons/10/package.json]: Done.

**Comment by jianhuang01** [comment]
@ibgreen  I didn't realize there is another "lesson 10" example under examples/wip/, I will remove them in this diff.

File: examples/lessons/10/app.js (hunk 1 above)
Comment (jianhuang01) [reply examples/lessons/10/app.js]: Done.

File: examples/lessons/10/app.js (hunk 4 above)
Comment (jianhuang01) [reply examples/lessons/10/app.js]: Figured out a solution to make example work on Safari without using "var", need wrap all variables with objects, and pass them to functions as parameters.

**Pull request merged**